src/com/goide/highlighting/exitpoint/GoBreakStatementExitPointHandler.java

```diff
@@ -0,0 +1,149 @@
+    selectionConsumer.consume(targets);
+  }
+
+  @Override
+  public void computeUsages(List<PsiElement> targets) {
+    final PsiElement breakStmtOwner;
+    if (myOwner != null) {
+      breakStmtOwner = myOwner;
+    }
+    else if (myBreakStatement != null) {
+      breakStmtOwner = getBreakStatementOwner(myBreakStatement);
+    }
+    else {
+      breakStmtOwner = null;
+    }
+    GoRecursiveVisitor visitor = new GoRecursiveVisitor() {
+      @Override
+      public void visitLabelDefinition(@NotNull GoLabelDefinition o) {
+        if (o == breakStmtOwner) {
+          addOccurrence(o);
+        }
+        super.visitLabelDefinition(o);
+      }
+
+      @Override
+      public void visitBreakStatement(@NotNull GoBreakStatement o) {
+        if (o == myBreakStatement || getBreakStatementOwner(o) == breakStmtOwner) {
+          addOccurrence(o);
+        }
+        super.visitBreakStatement(o);
+      }
+
+      @Override
+      public void visitSwitchStatement(@NotNull GoSwitchStatement o) {
+        if (o == breakStmtOwner) {
+          GoSwitchStart switchStart = o.getSwitchStart();
+          if (switchStart != null) {
+            addOccurrence(switchStart.getSwitch());
+          }
+        }
+        super.visitSwitchStatement(o);
+      }
+
+      @Override
+      public void visitForStatement(@NotNull GoForStatement o) {
+        if (o == breakStmtOwner) {
+          addOccurrence(o.getFor());
+        }
+        super.visitForStatement(o);
+      }
+
+      @Override
+      public void visitSelectStatement(@NotNull GoSelectStatement o) {
+        if (o == breakStmtOwner) {
+          addOccurrence(o.getSelect());
+        }
+        super.visitSelectStatement(o);
+      }
+    };
+    if (breakStmtOwner != null) {
+      PsiElement parent = breakStmtOwner.getParent();
+      if (parent instanceof GoCompositeElement) {
+        visitor.visitCompositeElement(((GoCompositeElement)parent));
+      }
+    }
+  }
+
+
+  @Nullable
+  private static PsiElement getBreakStatementOwner(@NotNull GoBreakStatement breakStmt) {
+    GoLabelRef label = breakStmt.getLabelRef();
+    if (label != null) {
+      return label.getReference().resolve();
+    }
+    GoCompositeElement breaksOutOf =
+      PsiTreeUtil.getParentOfType(breakStmt, GoSwitchStatement.class, GoForStatement.class, GoSelectStatement.class,
+                                  GoFunctionLit.class);
+    if (breaksOutOf instanceof GoFunctionLit) {
+      return null;
+    }
+    return breaksOutOf;
+  }
+
+  @Nullable
+  public static GoBreakStatementExitPointHandler createForElement(@NotNull Editor editor, @NotNull PsiFile file, @NotNull PsiElement element) {
+    PsiElement target =
+      PsiTreeUtil.getParentOfType(element, GoBreakStatement.class, GoSwitchStatement.class, GoSelectStatement.class, GoForStatement.class);
+    if (target == null) return null;
+    if (target instanceof GoBreakStatement) {
```

Comment:
Oh, I see, you need it for proper highlighting labeled breakStatement. Anyway, looks dirty, I don't see any sense not to pass owner to the handler in the first place.